### PR TITLE
Add "no public GH Issues please" request, past advisories link, bounty mention, scope link to security.md

### PR DESCRIPTION
**Pull request by joshtrichards**
## Summary

* Added request to _not_ report vulnerabilities in public GH issues
* Added missing and/or deeper links to relevant pages { existing security advisories, scope }
* Mentioned bounty program
* Reorganized to keep focus on reporting while also adding context helpful to other audiences
* Added some new headings to keep things easy to access

I was tempted to add the "No BS" stuff from [the HackerOne page](https://hackerone.com/nextcloud), but ultimately opted to keep this the more formal of the two since that is how it's been historically.

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -1,25 +1,62 @@
...
+If you believe you have found a security vulnerability that meets our definition of a security 
+vulnerability, please report is as described below.
+
+## Context
+
+Please review our [threat model and accepted risks](https://nextcloud.com/security/threat-model) to learn what 
+is currently considered a security vulnerability versus expected behavior. And review what is considered 
+[in scope or bounty eligible](https://hackerone.com/nextcloud/policy_scopes).
+
+You can expect a response within 24 hours in most cases.
```
Comment (nickvergessen) [inline SECURITY.md:14]:
```suggestion
```
It's mentioned in `What to Expect`. Should be good enough there.

File: SECURITY.md [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
+- Any other details you think are likely to be important
+
+### What to Expect
+
+You should receive an initial acknowledgement within 24 hours in most cases.
 
-A member of the security team will confirm the vulnerability, determine its impact, and develop a fix.
-The fix will be applied to the master branch, tested, and packaged in the next security release.
+A member of the security team will confirm the vulnerability, determine its impact, follow-up with any questions, 
+and coordinate a fix.
```
Comment (nickvergessen) [inline SECURITY.md:36]:
```suggestion
and coordinate the fix and publication.
```

File: SECURITY.md [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+
+The fix will be applied to the `master` branch, tested, and packaged in the next security release.
```
Comment (nickvergessen) [inline SECURITY.md:38]:
```suggestion
The fix will be applied to all applicable and still supported stable branches, tested, and packaged in the next security release.
```

File: SECURITY.md [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+community. 
+
+### Bug Bounties
+
+If you are reporting for a bug bounty, more complete reports can contribute to a higher bounty award. Details
+on past bounty ranges can be found at [hackerone.com/nextcloud](https://hackerone.com/nextcloud).
+
+## Existing Security Advisories
+
+Past advisories can be viewed at
```
Comment (nickvergessen) [inline SECURITY.md:50]:
```suggestion
Published security advisories for the Nextcloud Server, Clients and Apps can be viewed at
```

File: SECURITY.md [hunk 5]
```diff
@@ -1,25 +1,62 @@
...
+[https://github.com/nextcloud/security-advisories/security/advisories](https://github.com/nextcloud/security-advisories/security/advisories
+).
+
+## Supported Versions
+
+The latest three major release versions of Nextcloud are currently being supported with security updates.
```
Comment (nickvergessen) [inline SECURITY.md:56]:
```suggestion
Nextcloud Server major release versions are being supported with security updates for 1 year after their initial release.
```

**Review by nickvergessen — COMMENTED** [review]
Thanks for cleaning this one up.
Some small comments but good step forward :)

**Review by nickvergessen — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
Makes sense and LGTM. Should we apply the same changes to https://github.com/nextcloud/.github/blob/master/SECURITY.md?

**Pull request merged**